### ISSUE TITLE
Drawing independent lines without lineseries

### ISSUE THREAD

**Issue by ProgrammingLife** [tradingview/lightweight-charts#334]
When I draw a line (i.e. a resistance line or a support line) I use lineseries for that purpose. It's very uncomfortable when it comes to price scaling. Those lines also take a part in scaling when it shouldn't and all the plot scales wrong on shorter distances of the plot.

**Comment by timocov** [comment]
Can you provide example please? How did you create that lines? What do you expect for?

**Comment by ProgrammingLife** [comment]
I'm talking about creating adding lines for example. If those lines are too long they also scale by priceScale because those diagonal lines also have "prices" :) I'll make screenshot a bit later.

**Comment by ProgrammingLife** [comment]
Something like this. Those yellow dotted lines are just long lines which started at the left side of my candlestick series. But I don't want to see those yellow lines at this time range. I use autoScale for the priceScale but it doesn't help because yellow lines considers as price lines by the system when it isn't. They are just lines :) Is it clearer now?
![](https://i.imgur.com/q9G8nv4.png)

**Comment by timocov** [comment]
You want to see that lines on the same price scale as series' one (thus they'll have the same scale as the series), but you don't want them affects auto scaling, right?

**Comment by ProgrammingLife** [comment]
@timocov I wanna see just autoscaled candlestick series (by priceScale) no matter if those yellow line are visible or not at the time of current scale

**Comment by timocov** [comment]
@ProgrammingLife but these yellow lines should be scales _together_ with candlestick series, right?

**Comment by ProgrammingLife** [comment]
@timocov actually, yes :)

**Comment by timocov** [comment]
Awesome :) I got your point. Currently you can't do that. @eugene-korobko @ezhukovskiy @Nipheris what do you think about adding option `affectsAutoscale` or something like that to series options?

**Comment by eugene-korobko** [comment]
As I remember, we were working on custom price lines recently. 
We have a method `createPriceLine`
please refer [here](https://github.com/tradingview/lightweight-charts/blob/master/docs/series-basics.md)

**Comment by ProgrammingLife** [comment]
@eugene-korobko Well, it's not the price line through the whole plot. It can be a short line or a long line or even a diagonal line which createPriceLine method can't do.

**Comment by tolotrasamuel** [comment]
Is this feature in progress? I actually need it too. I just proposed it in an issue [396](https://github.com/tradingview/lightweight-charts/issues/396) 
I agree with @ProgrammingLife , we need something like the createPriceLine feature but not for the entire plot. 

We need these lines because they could be used as price levels, extra visuals, trendline...
The workaround I found is also to draw a line series with only two points. Here is an example:


```
const lineSeries2 = chart.addLineSeries();
lineSeries2.setData([
    { time: '2018-11-20', value: 300 },
    { time:  '2019-05-24', value: 300 },
    
]);
const lineSeries = chart.addLineSeries();
lineSeries.setData([
    { time: '2018-11-20', value: 173 },
    { time:  '2019-05-24', value: 173 }
]);
```
Here is the [jsfiddle](https://jsfiddle.net/doL1vsgj/3/) code.
The problem is that the line series affects the price scale when you auto-scale.

![image](https://user-images.githubusercontent.com/29230303/78700034-e0bd5500-7915-11ea-9ef1-d165afe6b04c.png)

The blue lines are the lines that I want to add.

This is unwanted because by changing the price scale, these lines makes the actual series (ohlc) hard to read.

The expected behavior is as follow:

![image](https://user-images.githubusercontent.com/29230303/78700122-02b6d780-7916-11ea-8cc9-3ad969f2d584.png)



The upper blue line can only between by manually changing the price scale or dragging the chart up.
The lower blue line just happened to be visible because it inside the candle price range


I found something similar already existing which is the `overlay` propriety but unfortunately, it creates a completely new price scale for the series. ( I do not know if that is the expected behavior or it's a bug but it produces something really messy )
![image](https://user-images.githubusercontent.com/29230303/78699979-ca16fe00-7915-11ea-8b03-a553fb808c31.png)





**My Propositions**

I propose either of three solutions:

1) Add an option `userSamePriceScale` to `OverlaySeriesSpecificOptions`

```
export interface OverlaySeriesSpecificOptions {
	overlay: true;
	scaleMargins?: PriceScaleMargins;
        useSamePriceScale: boolean; // Add this
}
```

This would force the overlay to use the same price scale as the non-overlay series already added , not create its own

2) Create a new type of API called `DrawingsApi` or something like.
It would be extended with `LineDrawing` or `CircleDrawing`.
And it would be added on the ChartApi as follow:

```
const lineDrawing = chart.addLineDrawing({
    from:{ time: '2018-11-20', value: 173 },
    to: { time: '2019-11-20', value: 173 },
});
```
        
3) Add `from` and `to` properties to `PriceLineOptions`

```
export interface PriceLineOptions {
	price: BarPrice;
	color: string;
	lineWidth: LineWidth;
	lineStyle: LineStyle;
	axisLabelVisible: boolean;
       from: Time; // Add this
       to: Time; // Add this
}
```

And use it like this:
`const priceLine = series.createPriceLine({ price: 80.0, from:'2018-11-20', to:'2019-11-20' });`

**Comment by timocov** [comment]
Hey everyone in this issue. Looks like we'll fix the issue in #392. We'll add kind of `autoscaleInfo` option to override default series' autoscale info getter so you are able to override it and just return `null` and we won't use that series in autoscaling.

**Comment by timocov** [comment]
I'm going to close the issue in favor #392. If you'll have any question or updates - feel free to re-open it.
